step.go

```diff
@@ -7,133 +7,129 @@ import (
+	log.Printf("- colorOnError: %s", configs.colorOnError)
 
-func printConfig(roomID, fromName, message, color, fromNameOnError, messageOnError, colorOnError, messageFormat string) {
-	logInfo("Configs:")
-	logDetails("token: ***")
-	logDetails("romm_id: %s", roomID)
-	logDetails("from_name: %s", fromName)
-	logDetails("color: %s", color)
-	logDetails("message: %s", message)
-	logDetails("from_name_on_error: %s", fromNameOnError)
-	logDetails("message_on_error: %s", messageOnError)
-	logDetails("color_on_error: %s", colorOnError)
-	logDetails("message_format: %s", messageFormat)
+	log.Printf("- messageFormat: %s", configs.messageFormat)
 }
 
 // -----------------------
 // --- Main
 // -----------------------
 
 func main() {
-	//
-	// Validate options
-	token := validateRequiredInput("auth_token")
-	roomID := validateRequiredInput("room_id")
-	fromName := validateRequiredInput("from_name")
-	message := validateRequiredInput("message")
-
-	//optional inputs
-	messageColor := validateInput("color", "yellow")
-
-	errorFromName := validateInput("from_name_on_error", fromName)
-	errorMessage := validateInput("message_on_error", message)
-	errorMessageColor := validateInput("color_on_error", messageColor)
-
-	messageFormat := validateInput("message_format", "text")
-
-	isBuildFailedMode := (os.Getenv("STEPLIB_BUILD_STATUS") != "0")
-	if isBuildFailedMode {
-		fromName = errorFromName
-		message = errorMessage
-		messageColor = errorMessageColor
-	}
 
-	printConfig(roomID, fromName, message, messageColor, errorFromName, errorMessage, errorMessageColor, messageFormat)
+	fmt.Println()
+
+	config := createConfigsModelFromEnvs()
+
+	config.print()
+
+	isFailed := (config.isBuildFailedMode != "0")
 
+	if isFailed {
+		config.fromName = config.fromNameOnError
+		config.message = config.messageOnError
+		config.color = config.colorOnError
+	}
+
+	
 	//
 	// Create request
-	logInfo("Performing request")
+	fmt.Println()
+	log.Infof("Performing request")
+	fmt.Println()
 
 	values := url.Values{
-		"room_id":        {roomID},
-		"from":           {fromName},
-		"message":        {message},
-		"color":          {messageColor},
-		"message_format": {messageFormat},
+		"room_id":        {config.roomID},
+		"from":           {config.fromName},
+		"message":        {config.message},
+		"color":          {config.color},
+		"message_format": {config.messageFormat},
 	}
+
 	valuesReader := *strings.NewReader(values.Encode())
 
-	url := baseURL + "/rooms/message?auth_token=" + token
+	url := baseURL + "/rooms/message?auth_token=" + config.token
 
 	request, err := http.NewRequest("POST", url, &valuesReader)
+
 	if err != nil {
-		logFail("Failed to create request, error: %s", err)
+		log.Errorf("failed to perform request, error: %s", err)
+		os.Exit(1)
 	}
+
 	request.Header.Add("Accept", "application/json")
 	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 
 	client := &http.Client{}
 	response, requestErr := client.Do(request)
```

Comment:
Could you please rename `requestErr` to `err` and handle the error here? If error is not nil, the body will be empty.

Please take a look, how we handle request in [HockeyApp step](https://github.com/bitrise-io/steps-hockeyapp-deploy/blob/master/main.go#L182)